icelake/src/types/in_memory.rs

```diff
@@ -2007,6 +2005,253 @@ impl Snapshot {
     }
 }
 
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+pub struct SnapshotSummary {
```

Comment:
I prefer to only store a `HashMap<String, String>` here.  The snapshot summary not only contains metrics, but may also properties. It's ok to store the metrics in `SnapshotSummaryBulder` for easier manipulation.